File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    public List<TbBrand> findAll();

    public PageResult findByPage(Integer pageNum, Integer pageSize);

    void addBrand(TbBrand tbBrand) throws Exception;

    public TbBrand findOne(long id);

    public void updateById(TbBrand tbBrand) throws Exception;

    public void deleteById(long[] ids) throws  Exception;

    PageResult findByPageWithCondition(Integer pageNum, Integer pageSize, TbBrand tbBrand);

    List<Map> selectOptionList();

}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService", function ($http) {

    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }

    this.findPage=function (pageNum,pageSize,searchEntity) {
        return $http.post("../brand/findByPage.do?pageNum="+pageNum+"&pageSize="+pageSize+"",searchEntity);
    }

    this.addBrand=function (entity) {
        return $http.post("../brand/addBrand.do", entity);
    }

    this.updateById=function (entity) {
        return $http.post("../brand/updateById.do", entity);
    }

    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id+"");
    }

    this.delById=function (ids) {
        return $http.post("../brand/deleteById.do", ids);
    }

    this.findOptionList=function () {
        return $http.get("../brand/selectOptionList.do");
    }
    //搜索
    this.search=function(page,rows,searchEntity){
        return $http.post('../brand/search.do?page='+page+"&rows="+rows, searchEntity);
    }

})